lib/Service/Mail/Mailer.hs

```diff
@@ -4,64 +4,148 @@ module Service.Mail.Mailer
 import qualified Data.UUID as U
 import qualified Network.HaskellNet.Auth as Auth
 import qualified Network.HaskellNet.SMTP as SMTP
 import qualified Network.HaskellNet.SMTP.SSL as SMTPSSL
 import qualified Network.Mail.Mime as MIME
 import qualified Network.Mail.SMTP as SMTPMail
+import qualified Network.Mime as MIME
+import System.Directory (listDirectory)
+import System.FilePath (takeFileName)
 
 import Constant.Component
 import LensesConfig
 import Localization
 import Model.Context.AppContext
 import Model.User.User
 import Util.Logger
+import Util.Template (loadAndRender)
 
-sendRegistrationConfirmationMail :: Email -> U.UUID -> String -> AppContextM ()
-sendRegistrationConfirmationMail email userId hash = do
+sendRegistrationConfirmationMail :: User -> U.UUID -> String -> AppContextM ()
+sendRegistrationConfirmationMail user userId hash = do
   dswConfig <- asks _appContextConfig
   let clientAddress = dswConfig ^. clientConfig . address
-      clientLink = clientAddress ++ "/signup-confirmation/" ++ U.toString userId ++ "/" ++ hash
-      link = "<a href=\"" ++ clientLink ++ "\">here</a>"
+      activationLink = clientAddress ++ "/signup-confirmation/" ++ U.toString userId ++ "/" ++ hash
       mailName = dswConfig ^. mail . name
       subject = TL.pack $ mailName ++ ": Confirmation Email"
-      body = TL.pack $ "Hi! For account activation you have to click " ++ link ++ "! " ++ mailName ++ " Team"
-  sendEmail [email] subject [SMTPMail.htmlPart body]
+      additionals = [("activationLink", (Aeson.String $ T.pack activationLink))]
+      context = makeMailContext mailName clientAddress user additionals
+  parts <- loadMailTemplateParts RegistrationConfirmation context
+  sendEmail [user ^. email] subject parts
 
-sendRegistrationCreatedAnalyticsMail :: String -> String -> Email -> AppContextM ()
-sendRegistrationCreatedAnalyticsMail uName uSurname uEmail = do
+sendRegistrationCreatedAnalyticsMail :: User -> AppContextM ()
+sendRegistrationCreatedAnalyticsMail user = do
   dswConfig <- asks _appContextConfig
-  let analyticsAddress = dswConfig ^. analytics . email
+  let clientAddress = dswConfig ^. clientConfig . address
+      analyticsAddress = dswConfig ^. analytics . email
       mailName = dswConfig ^. mail . name
       subject = TL.pack $ mailName ++ ": New user"
-      body =
-        TL.pack $ "Hi! We have a new user (" ++ uName ++ " " ++ uSurname ++ ", " ++ uEmail ++ ") in our Wizard! " ++
-        mailName ++
-        " Team"
-  sendEmail [analyticsAddress] subject [SMTPMail.htmlPart body]
-
-sendResetPasswordMail :: Email -> U.UUID -> String -> AppContextM ()
-sendResetPasswordMail email userId hash = do
+      context = makeMailContext mailName clientAddress user []
+  parts <- loadMailTemplateParts RegistrationCreatedAnalytics context
+  sendEmail [analyticsAddress] subject parts
+
+sendResetPasswordMail :: User -> U.UUID -> String -> AppContextM ()
+sendResetPasswordMail user userId hash = do
```

Comment:
Can we take `userId` from `user` object?

Reply:
https://github.com/ds-wizard/dsw-server/pull/45/commits/38a98bf1eb7538a2b961fb9cca15b0760ae8fb7c